internal/integrations/utils.go

```diff
@@ -0,0 +1,194 @@
+		for _, pluginV1Instance := range v3Configuration.Instances {
+			if commandName == pluginV1Instance.Command {
+				integrationInstance := populateConfigEntry(pluginV1Instance, pluginV1Command)
+				v4Config.Integrations = append(v4Config.Integrations, integrationInstance)
+			}
+		}
+	}
+
+	return v4Config, nil
+}
+
+func populateConfigEntry(pluginV1Instance *PluginV1Instance, pluginV1Command *PluginV1Command) ConfigEntry {
+	configEntry := ConfigEntry{}
+	if len(pluginV1Command.Command) == 0 {
+		return configEntry
+	}
+
+	executable := pluginV1Command.Command[0]
+	binaryName := filepath.Base(executable)
+	configEntry.InstanceName = strings.TrimSuffix(binaryName, exeSuffix)
+	configEntry.Interval = fmt.Sprintf("%ds", pluginV1Command.Interval)
+	configEntry.Labels = pluginV1Instance.Labels
+	configEntry.User = pluginV1Instance.IntegrationUser
+	configEntry.InventorySource = pluginV1Command.Prefix
+	configEntry.Env = map[string]string{}
+	for k, v := range pluginV1Instance.Arguments {
+		configEntry.Env[strings.ToUpper(k)] = v
+	}
+
+	// Please notice that this is a simplification. If it is an absolute path we are adding it to the exec
+	// if is a relative path or a integration name, we are assuming it is a standard integration included into the path
+	if filepath.IsAbs(executable) {
+		configEntry.Exec = pluginV1Command.Command
+	} else {
+		buildCLIArgs(pluginV1Command, &configEntry)
+	}
+	return configEntry
+}
+
+func buildCLIArgs(pluginV1Command *PluginV1Command, configEntry *ConfigEntry) {
+	for index, arg := range pluginV1Command.Command {
+		if index == 0 {
+			// the first arg in command is the binary name
+			continue
+		}
+
+		sanitized := strings.TrimPrefix(arg, prefixArg)
+		sanitized = strings.TrimPrefix(sanitized, prefixArgShort)
+		if sanitized == metricArg || sanitized == inventoryArg || sanitized == eventsArg {
+			configEntry.Env[strings.ToUpper(sanitized)] = "true"
+		} else {
+			configEntry.CLIArgs = append(configEntry.CLIArgs, arg)
+		}
+	}
+}
+
+func writeOutput(pathOutput string, v4Config *v4, definitionBytes []byte, configurationBytes []byte) error {
+	if v4Config == nil {
+		return fmt.Errorf("v4Config pointer is nil")
+	}
+
+	file, err := os.OpenFile(pathOutput, os.O_RDWR|os.O_CREATE|syscall.O_TRUNC, 0666)
+	if err != nil {
+		return fmt.Errorf("opening File %s, %w", pathOutput, err)
+	}
+	defer file.Close()
+
+	err = writeV4Config(v4Config, file)
+	if err != nil {
+		return fmt.Errorf("writing v4 config, %w", err)
+	}
+
+	err = writeTextAsComment(file, string(definitionBytes))
+	if err != nil {
+		return fmt.Errorf("adding old definition as comment, %w", err)
+	}
+
+	err = writeTextAsComment(file, string(configurationBytes))
+	if err != nil {
+		return fmt.Errorf("adding old configuration as comment, %w", err)
+	}
+
+	return nil
+}
+
+func writeV4Config(v4Config *v4, file *os.File) error {
+	// see https://github.com/go-yaml/yaml/commit/7649d4548cb53a614db133b2a8ac1f31859dda8c
+	yaml.FutureLineWrap()
```

Comment:
Won't be needed with `v3`, according to the link

Reply:
v2 is already used across the code, I did not want to update a dependency we do not own or import two versions of yaml